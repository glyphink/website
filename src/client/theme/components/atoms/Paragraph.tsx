/**
 * @file Atoms - Paragraph
 * @module website/components/atoms/Paragraph
 */

import type { Props } from '@glyphink/website/theme'
import {
  forwardRef,
  type ForwardedRef,
  type ForwardRefExoticComponent,
  type JSX
} from 'react'
import type ParagraphProps from './Paragraph.props.mts'

/**
 * A paragraph.
 *
 * @see https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/p
 * @see {@linkcode ForwardRefExoticComponent}
 * @see {@linkcode ParagraphProps}
 *
 * @category
 *  atoms
 *
 * @const {ForwardRefExoticComponent<ParagraphProps>} Paragraph
 */
const Paragraph: ForwardRefExoticComponent<ParagraphProps> = forwardRef<
  HTMLParagraphElement,
  ParagraphProps
>(
  /**
   * @param {Props<ParagraphProps>} props
   *  Component props
   * @param {ForwardedRef<HTMLParagraphElement>} ref
   *  Forwarded ref
   * @return {JSX.Element}
   *  JSX `<p>` element
   */
  (
    props: Props<ParagraphProps>,
    ref: ForwardedRef<HTMLParagraphElement>
  ): JSX.Element => {
    return <p {...props} ref={ref} />
  }
)

Paragraph.displayName = 'Paragraph'

export default Paragraph
