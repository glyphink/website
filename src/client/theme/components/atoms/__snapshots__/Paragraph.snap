// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:theme/components/atoms/Paragraph > should render <p> 1`] = `
<p>
  the quick brown fox jumped over the lazy dog
</p>
`;
